optaweb-employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/roster/RosterController.java

```diff
@@ -108,6 +120,36 @@ public RosterController(RosterService rosterService) {
                                                                         spots), HttpStatus.OK);
     }
 
+    @ApiOperation("Get a shift roster view between two dates for a subset of the spots as an excel file")
+    @GetMapping(value = "/shiftRosterView/excel",
+            produces = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet")
+    public ResponseEntity<byte[]> getShiftRosterViewAsExcel(@PathVariable @Min(0) Integer tenantId,
+                                                            @RequestParam(name = "startDate")
+                                                                    String startDateString,
+                                                            @RequestParam(name = "endDate") String endDateString,
+                                                            @RequestParam(name = "spotList") String spotListString) {
+        Set<Long> spotIdSet = Arrays.stream(spotListString.split(",")).map(Long::parseLong)
+                .collect(Collectors.toSet());
+        List<Spot> spotList = spotRepository.findAllByTenantId(tenantId, PageRequest.of(0, Integer.MAX_VALUE))
+                .stream().filter(s -> spotIdSet.contains(s.getId()))
+                .collect(Collectors.toList());
```

Comment:
Important piece of application logic that should have a test coverage. Also, filtering spots by ID is outside of the controller's responsibility. This should be done by the `RosterService`.

`RosterServiceTest`:
- _given_ **spots** in the repository and a **list** of spot IDs,
- _when_ roster view is requested for the given **list**,
- _then_ roster view is returned with spots from an intersecion of IDs in the list and the repository.

Reply:
It appears `RosterService` does not have unit tests, unlike the other services. Out of scope for this PR (This PR will only have the controller tests as a result; the unit tests for RosterService will come in a different PR)